packages/bundler/src/modules/MempoolManager.ts

```diff
@@ -137,6 +137,27 @@ export class MempoolManager {
     }
   }
 
+  private checkMultipleRolesViolation (userOp: UserOperation): void {
+    const knownEntities = this.getKnownEntities()
+    requireCond(
+      !knownEntities.includes(userOp.sender.toLowerCase()),
+      `The sender address "${userOp.sender}" is used as a different entity in another UserOperation currently in mempool`,
+      ValidationErrors.OpcodeValidation
+    )
+
+    const knownSenders = this.getKnownSenders()
+    const paymaster = getAddr(userOp.paymasterAndData)?.toLowerCase()
+    const factory = getAddr(userOp.initCode)?.toLowerCase()
+
+    const isPaymasterSenderViolation = knownSenders.includes(paymaster?.toLowerCase() ?? '')
+    const isFactorySenderViolation = knownSenders.includes(factory?.toLowerCase() ?? '')
+    requireCond(
```

Comment:
instead of 2 booleans, and a combined condition it looks like 2 separate conditions would work better
(e.g.
requireCond(!knownSenders.includes(factory), "factory is already a sender")
requireCond(!knownSenders.includes(paymaster), "paymaster is already a sender")